Cargo.toml

```diff
@@ -13,25 +13,26 @@ readme = "README.md"
 rust-version = "1.81"
 
 [dependencies]
-num-bigint = { version = "0.8.2", features = ["i128", "prime", "zeroize"], default-features = false, package = "num-bigint-dig" }
-num-traits = { version = "0.2.9", default-features = false, features = ["libm"] }
-num-integer = { version = "0.1.39", default-features = false }
 rand_core = { version = "0.6.4", default-features = false }
 const-oid = { version = "0.10.0-rc.3", default-features = false }
-subtle = { version = "2.1.1", default-features = false }
+subtle = { version = "2.6.1", default-features = false }
 digest = { version = "=0.11.0-pre.9", default-features = false, features = ["alloc", "oid"] }
 pkcs1 = { version = "0.8.0-rc.0", default-features = false, features = ["alloc", "pkcs8"] }
 pkcs8 = { version = "0.11.0-rc.0", default-features = false, features = ["alloc"] }
 signature = { version = "=2.3.0-pre.4", default-features = false, features = ["alloc", "digest", "rand_core"] }
 spki = { version = "0.8.0-rc.1", default-features = false, features = ["alloc"] }
 zeroize = { version = "1.5", features = ["alloc"] }
+crypto-bigint = { version = "0.6.0-rc.6", default-features = false, features = ["zeroize", "alloc"] }
+crypto-primes = { version = "0.6.0-pre.2", default-features = false }
+libm = "0.2"
```

Comment:
IMO it would be nice to avoid this, especially if its only used for `no_std` prime counting

Reply:
If all you need from there is `logf(x)`, you can probably just approximate it with `x.bits() / log2(e)`, pre-calculating `log2(e)`.

Reply:
Or even take `log2(e) == 1` if the approximation allows it :)

Reply:
`bits()` seems to not be around either..

Reply:
Well, the only instance of `logf()` usage is
```
let prime_limit = (1u64 << (bit_size / nprimes) as u64) as f64;
let mut pi = prime_limit / (logf(prime_limit) - 1f64);
```
So you can just do
```
let log2_e = 1.4426950408889634;
let mut pi = prime_limit / ((bit_size / nprimes) / log2_e  - 1); // with whatever type conversions required
```
Does that work?

Reply:
I couldn't get it to work, but I did find a nice and short ln implementation on the internet that I added. and now no more `libm` dependency

Reply:
the results are not even close, they are completely different numbers

Reply:
That's pretty strange. Seems to work for me:
```rust
fn main() {
    let bit_size = 1024;
    let nprimes = 42;

    let prime_limit = (1u64 << (bit_size / nprimes) as u64) as f64;
    let pi_ref = prime_limit / (prime_limit.ln() - 1f64);

    println!("{pi_ref}");

    let log2_e = 1.4426950408889634;
    let pi_test = prime_limit / ((bit_size / nprimes) as f64 / log2_e  - 1f64); // with whatever type conversions required

    println!("{pi_test}");
}
```
```
1073018.5351041532
1073018.5351041532
```

Reply:
@dignifiedquire